linetools/spectra/io.py

```diff
@@ -199,7 +202,11 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, multi_ivar=False,
 
     # Add in the header
     #xspec1d.head = head0
-    xspec1d.meta['headers'][0] = head0
+    if head_ext == 0:
```

Comment:
`head_spec` would be your current `head = hdulist[head_ext].header` (apologies for being pedantic...)